spec.html

```diff
@@ -10543,7 +10545,8 @@ <h1>Host Hooks</h1>
 
       <emu-clause id="sec-host-cleanup-finalization-registry" aoid="HostEnqueueFinalizationRegistryCleanupJob">
         <h1>HostEnqueueFinalizationRegistryCleanupJob ( _finalizationRegistry_ )</h1>
-        <p>The abstract operation HostEnqueueFinalizationRegistryCleanupJob takes argument _finalizationRegistry_ (a FinalizationRegistry). HostEnqueueFinalizationRegistryCleanupJob is an implementation-defined abstract operation that is expected to call CleanupFinalizationRegistry(_finalizationRegistry_) at some point in the future, if possible. The host's responsibility is to make this call at a time which does not interrupt synchronous ECMAScript code execution.</p>
+        <p>The abstract operation HostEnqueueFinalizationRegistryCleanupJob takes argument _finalizationRegistry_ (a FinalizationRegistry). HostEnqueueFinalizationRegistryCleanupJob is an implementation-defined abstract operation that is expected to call CleanupFinalizationRegistry(_finalizationRegistry_) at some point in the future, if possible.</p>
+        <p>An implementation of HostEnqueueFinalizationRegistryCleanupJob must conform to the requirements in <emu-xref href="#sec-jobs"></emu-xref>.</p>
```

Comment:
> The host's responsibility is to make this call at a time which does not interrupt synchronous ECMAScript code execution.

At first glance, I thought the new requirement on host hooks obviated this statement. On second thought, it doesn't seem like it applies to the behavior of an abstract operation invoked after the host hook has returned. Is there some other justification for removing the statement?

Reply:
The intention here was the "must not interrupt synchronous execution" is more precisely captured by the requirements in `#sec-jobs`, specifically the stuff around waiting for the execution context stack to be empty. No job queued by a job-queuing host defined AO in ecma262 is allowed to interrupt synchronous execution.

Reply:
Hm, a strict reading of the `#sec-jobs` restriction only applies to Job Abstract Closures, which HostEnqueueFinalizationRegistryCleanupJob explicitly spells out. I'll spell that out so it's clear that it applies.